docs/manual/src/proc_macro/functions.md

```diff
@@ -0,0 +1,88 @@
+# Functions, Constructors, Methods
+
+Functions are exported to the namespace with the `#[uniffi::export]` attribute
+
+```rust
+#[uniffi::export]
+fn hello_world() -> String {
+    "Hello World!".to_owned()
+}
+```
+
+All owned [builtin types](../types/builtin_types.md) and user-defined types can be used as arguments
+and return types.
+
+Arguments and receivers can also be references to these types, for example:
+
+```rust
+// Input data types as references
+#[uniffi::export]
+fn process_data(a: &MyRecord, b: &MyEnum, c: &Option<MyRecord>) {
+    ...
+}
+
+## Default values
```

Comment:
Before this (and after the general section above), I think it would be good to have a reference to the part of `interfaces.md` that tells you that `#[uniffi::export]` on an `impl` block must be for an interface type.

Reply:
I think this is what you meant!